tempodb/backend/azure/azure_helpers.go

```diff
@@ -9,40 +9,31 @@ import (
 	"strings"
 	"time"
 
-	"github.com/grafana/tempo/tempodb/backend/instrumentation"
-
-	"github.com/Azure/azure-pipeline-go/pipeline"
-	blob "github.com/Azure/azure-storage-blob-go/azblob"
-	"github.com/Azure/go-autorest/autorest/adal"
-	"github.com/Azure/go-autorest/autorest/azure"
-	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
 	"github.com/cristalhq/hedgedhttp"
+
+	"github.com/grafana/tempo/tempodb/backend/instrumentation"
 )
 
 const (
 	maxRetries = 1
 )
 
-var defaultAuthFunctions = authFunctions{
-	NewOAuthConfigFunc: adal.NewOAuthConfig,
-	NewServicePrincipalTokenFromFederatedTokenFunc: adal.NewServicePrincipalTokenFromFederatedToken,
-}
-
-type authFunctions struct {
-	NewOAuthConfigFunc                             func(activeDirectoryEndpoint, tenantID string) (*adal.OAuthConfig, error)
-	NewServicePrincipalTokenFromFederatedTokenFunc func(oauthConfig adal.OAuthConfig, clientID string, jwt string, resource string, callbacks ...adal.TokenRefreshCallback) (*adal.ServicePrincipalToken, error)
-}
-
-func GetContainerURL(ctx context.Context, cfg *Config, hedge bool) (blob.ContainerURL, error) {
+func getContainerClient(ctx context.Context, cfg *Config, hedge bool) (container.Client, error) {
 	var err error
-	var p pipeline.Pipeline
 
-	retryOptions := blob.RetryOptions{
-		MaxTries: int32(maxRetries),
-		Policy:   blob.RetryPolicyExponential,
+	retry := policy.RetryOptions{
+		MaxRetries:    int32(maxRetries) - 1,
```

Comment:
This configuration option was called "MaxTries" in the old SDK. It stated:
> "A value of 1 means 1 try and no retries."

Since the new option is "MaxRetries" (retries instead of tries), we need to set it to (previous value - 1) to keep the same behaviour.

This seems rather clumsy; a well-meaning engineer might update `maxRetries` to `0`, in which case we would be setting `MaxRetries` here to `-1` which I imagine would cause all sorts of errors.

We could hardcode the value to `0` here.

In fact, I wonder if this was an oversight of the implementation of the old SDK. With the old SDK, we were using `maxRetries` to set both `retryOptions.MaxTries` _and_ `DownloadFromBlobOptions.MaxRetryRequests`:
```go
	retryOptions := blob.RetryOptions{
		MaxTries: int32(maxRetries),
		Policy:   blob.RetryPolicyExponential,
	}
```

```go
		destBuffer, blob.DownloadFromBlobOptions{
			BlockSize:   blob.BlobDefaultDownloadBlockSize,
			Parallelism: maxParallelism,
			Progress:    nil,
			RetryReaderOptionsPerBlock: blob.RetryReaderOptions{
				MaxRetryRequests: maxRetries,
			},
		},
```

Is the desired behaviour that we use the same amount of retries across the board? If it is, then that would be a functional change, which I'm trying to stay away from in this pull request.

Reply:
so when constructing the "url" we were using max retries incorrectly and actually setting it to 0 retries. however, when calling `DownloadBlobToBuffer` we were presumably using it correctly and signalling we wanted 1 retry. naturally we have no idea how these two values interact in the real world.

since the intent seemed to be that we always had 1 retry, i think we should just set:
```
MaxRetries: int32(maxRetries),
```
and forget the -1. Likely worth noting in the changelog since it may cause behavioral changes in Tempo.

Reply:
> since the intent seemed to be that we always had 1 retry, i think we should just set:
> ```
> MaxRetries: int32(maxRetries),
> ```

Roger that.

> Likely worth noting in the changelog since it may cause behavioral changes in Tempo.

Good idea.

Actions taken:
* We now set `MaxRetries` to `1` instead of `0`.
* Added changelog `ENHANCEMENT` entry for the change